src/main/java/cz/educanet/collections/LinearProbing.java

```diff
@@ -0,0 +1,66 @@
+package cz.educanet.collections;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class LinearProbing<T> implements Set<T> {
+    private Object[] table; // Úložiště pro prvky
+    private int size;
+
+    public LinearProbing(int capacity) {
+        table = new Object[capacity];
+        size = 0;
+    }
+
+    // Implementace hašovací funkce pro určení indexu pro prvek
+    private int hash(T item) {
+        return item.hashCode() % table.length;
+    }
+
+    @Override
+    public boolean add(T item) {
+        int index = hash(item);
+        while (table[index] != null) {
+            if (table[index].equals(item)) {
+                return false; // Prvek již existuje, nelze ho přidat
+            }
+            index = (index + 1) ; // posun o 1
+        }
+        table[index] = item;
+        size++;
+        return true;
+    }
+    @Override
+    public boolean remove(T item) {
+        int index = hash(item);
+
+        while (table[index] != null) {
+            if (table[index].equals(item)) {
+                table[index] = null;
+                size--;
+                return true;
+            }
+            index = (index + 1);
+        }
+        return false; // Prvek nebyl nalezen
+    }
+
+    @Override
+    public boolean contains(T item) {
+        int index = hash(item);
+
+        while (table[index] != null) {
```

Comment:
pokud bych přidal 3 prvky, které se mají uložit na stejné místo, (např. 1001, 2001 a 3001). pak ten prostřední (2001) odebral a chtěl vědět, jestli obsahuje ten 3 (3001) tak vrátíš false, protože v tom poli by to bylo [1001, null, 3001]. Takže by tato metoda vrátila špatnou hodnotu